### ISSUE TITLE
Allow (Objective-)C(++) dynamic_frameworks

### ISSUE THREAD

**Issue by cpsauer** [bazelbuild/rules_apple#1123]
It's great to have [ios|watchos|tvos]_dynamic_frameworks, and I was excited to switch to them from ios_framework.

However, each seems to assume the code they bundle is written in Swift. In particular, swapping in ios_dynamic_framework for an ios_framework written in Objective-C, leads to:
`Error in fail:     error: Swift dynamic frameworks expect a single swift_library dependency.`

And from looking at [the definition](https://github.com/bazelbuild/rules_apple/blob/5d835252d2bc2d9535387dd73077a659d1b5b7b4/apple/internal/ios_rules.bzl), it's easy to see that this is coming from the assumption that the dynamic framework is in Swift. What's not obvious to me is whether the the dynamic_frameworks' current implementations are actually that Swift specific, or whether the check just makes more general code unnecessarily specific. 

TL;DR: Could we make [ios|watchos|tvos]_dynamic_frameworks language agnostic--and therefore support (Objective-)C(++), like the rest of the rules?

**Comment by brentleyjones** [comment]
cc: @mccorkill1

**Comment by mccorkill1** [comment]
Hi @cpsauer I'm happy to hear you want to use the dynamic_framework rules! When I wrote the rules I was specifically targeting swift_library because that was how ios_static_framework worked and I took a lot of code from there. I tried removing the check and testing with an objc_library target, but it didn't work quite right. I'm much more familiar with swift and since that is our primary use case, adding Objective-C support to the rule got deprioritized while we adopt the rule internally. I would love to see these rules support Objective-C, but I won't get to adding that functionality for a while. If you want to contribute it yourself I'd be happy to review a PR or answer any questions you might have during the dev process.

**Comment by cpsauer** [comment]
Found myself coming back here while looking into the new xcframework support.

Given apple's changes, probably not worth anyone investing more time into this (deprecated) way of doing things? What do you guys think? Should we close this down?

[And does anyone happen to know if apple_xcframework supports Obj-C (or C++) code?]

**Comment by thii** [comment]
It does. There are some examples in the unit tests:

https://github.com/bazelbuild/rules_apple/blob/f734e551f597266a300f883379bbbea9c47778e0/test/starlark_tests/targets_under_test/apple/BUILD#L257-L276

**Comment by cpsauer** [comment]
Spectacular! Thanks, @thii.

Shall we close, seeing as how xcframework is the way to distribute going forward?

**Comment by thii** [comment]
Yes, you can always post-process the `.xcframework` to get a single-platform `.framework` if you strictly need it for anything.

**Comment by louwers** [comment]
This limitation should at least be documented.

An `apple_xcframework` cannot be used by `ios_application` so I don't think it is a good alternative.
